lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
+  @spec gcd(0, 0) :: 0
+  @spec gcd(int, int) :: int when int: pos_integer
```

Comment:
No, I am not saying that. You are forgetting the subtyping relationship.

Imagine this: type is `number()`, which means it can be `integer()` or `float()`. When you have this:

    fun(a, a) when a: number()

Then the first argument and the second argument will be either an integer or a float. This won't be a valid call: `fun(1, 1.0)`.

But if you have this `fun(number(), number())`, then `fun(1, 1.0)` is a valid call.